microservice/src/main/java/nearsoft/skytouch/microservice/repository/impl/ProductRepositoryImpl.java

```diff
@@ -0,0 +1,65 @@
+package nearsoft.skytouch.microservice.repository.impl;
+
+import nearsoft.skytouch.common.model.Product;
+import nearsoft.skytouch.microservice.repository.ProductRepository;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.ParameterMode;
+import javax.persistence.PersistenceContext;
+import javax.persistence.StoredProcedureQuery;
+import javax.transaction.Transactional;
+import java.util.LinkedList;
+import java.util.List;
+
+@Repository
+public class ProductRepositoryImpl implements ProductRepository {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Transactional
+    @Override
+    public Product storeProduct(Product product) {
+        StoredProcedureQuery query = entityManager.createStoredProcedureQuery("addproduct");
```

Comment:
If you're using Spring Data, can't you use its features to execute SPs?.

Reply:
Yes but i was having some problems with that, i found this way to do that quickly, but i'm thinking on change it